vita3k/codec/src/mjpeg.cpp

```diff
@@ -168,6 +173,7 @@ int convert_yuv_to_jpeg(const uint8_t *yuv, uint8_t *jpeg, uint32_t width, uint3
     frame->format = context->pix_fmt;
     frame->width = context->width;
     frame->height = context->height;
+    frame->quality = FF_QP2LAMBDA * ((compress_ratio) * (31 - 1) / 255 + 1);
```

Comment:
FF_QP2LAMBDA is a bit arbitrary (related to H.263 quality), is there a reason you didn't use FF_LAMBDA_MAX? Does it compress the image too much with the default compression?

Reply:
Unlike the annotation, FFMPEG's FF_QP2LAMBDA is used to convert FFMPEG's qscale option to internal value without distinguishing between encodings.
(and many encoders that don't need the value multiplied by FF_QP2LABMDA, divide that value by FF_QP2LAMBDA again).

https://github.com/FFmpeg/FFmpeg/blob/af3f12078563d8119bd59771793994f651842ff2/fftools/ffmpeg_mux_init.c#L1219-L1222

Current code attempts to mimic the behavior of the FFMPEG CLI's qscale option. MJPEG's qscale is in the range of 1-31, and a 1 would be treated the same as a 2.
However, FFMPEG is overly complicated, so it is possible that my understanding is wrong.

Reply:
Ok, thank you for the explanation.